pkg/controller/sts_handlers.go

```diff
@@ -116,10 +116,25 @@ func (c *Controller) AssumeRoleWithWebIdentityHandler(w http.ResponseWriter, r *
 		return
 	}
 
+	isSSTSAudience := false
+	for _, audience := range saAuthResult.Status.Audiences {
+		if audience == TokenReviewAudience {
+			isSSTSAudience = true
+		}
+	}
+
+	if !isSSTSAudience {
+		if !saAuthResult.Status.Authenticated {
```

Comment:
This is essentially the same as `!isSTSAudience && !saAuthResult.Status.Authenticated`, but if the STS audience isn't listed, then it should return the errror. I guess the `!saAuthResult.Status.Authenticated` check should be removed (it will be error out on the next check).

Reply:
Logging here i `"Access denied: Invalid Token, audience 'sts.min.io' not found` is helpful to easely troubleshoot why is the token rejected, I'd rather keep this granular error case.